tools/file_packager.py

```diff
@@ -650,20 +673,22 @@ def generate_js(data_target, data_files, metadata):
           code += ('''Module['FS_createPath'](%s, %s, true, true);\n'''
                    % (json.dumps('/' + '/'.join(parts[:i])), json.dumps(parts[i])))
           partial_dirs.append(partial)
-
-  if options.has_preloaded:
+  has_preloaded = any(f.mode == 'preload' for f in data_files)
+  has_embedded = any(f.mode == 'embed' for f in data_files)
```

Comment:
Can these values change from above to here?   Surely the the mode of the data file is fixed and doesn't change?

Reply:
How about we disallow combining the two modes?